internal/cli/fetch_for_space.go

```diff
@@ -0,0 +1,285 @@
+	return ArgumentsDetails{
+		"space": "space to fetch",
+	}
+}
+
+func (c *FetchForSpace) ParseArgs(args ...string) (Arguments, error) {
+	return c.parseArgs(c, args...)
+}
+
+func (c *FetchForSpace) Execute(args Arguments) error {
+	start := time.Now()
+	log.Info().Msg("FetchForSpace started")
+	ctx := context.Background()
+
+	space, err := c.getSpace(args)
+	if err != nil {
+		return err
+	}
+
+	err = c.fetchSpace(ctx, space)
+	if err != nil {
+		return err
+	}
+
+	proposalIds, err := c.fetchProposals(ctx, space)
+	if err != nil {
+		return err
+	}
+
+	for i := range proposalIds {
+		if err := c.fetchProposalVotes(ctx, proposalIds[i]); err != nil {
+			return err
+		}
+	}
+
+	log.Info().Msgf("FetchForSpace finished. Took: %v", time.Since(start))
+
+	return nil
+}
+
+func (c *FetchForSpace) getSpace(args Arguments) (string, error) {
+	src := args.Get("space")
+	if src == "" {
+		return "", errors.New("space is required")
+	}
+
+	return src, nil
+}
+
+func (c *FetchForSpace) fetchSpace(ctx context.Context, space string) error {
+	s, err := c.Sdk.GetSpaceByID(ctx, space)
+	if err != nil {
+		return err
+	}
+	marshaled, err := json.Marshal(s)
+	if err != nil {
+		return err
+	}
+
+	sd := db.Space{
+		ID:        s.ID,
+		CreatedAt: time.Now(),
+		UpdatedAt: time.Now(),
+		Snapshot:  marshaled,
+	}
+
+	if err := c.Spaces.Upsert(&sd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *FetchForSpace) fetchProposals(ctx context.Context, space string) ([]string, error) {
+	opts := []snapshot.ListProposalOption{snapshot.ListProposalWithOrderBy("created", client.OrderDirectionAsc),
+		snapshot.ListProposalWithSpacesFilter(space)}
+	proposals, err := c.Sdk.ListProposal(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+	proposalIds := make([]string, 0)
+	for _, proposal := range proposals {
+		marshaled, err := json.Marshal(proposal)
+		if err != nil {
+			return proposalIds, err
+		}
+
+		p := db.Proposal{
+			ID:        proposal.ID,
+			SpaceID:   proposal.GetSpace().GetID(),
+			CreatedAt: time.Unix(proposal.GetCreated(), 0),
+			UpdatedAt: time.Now(),
+			Snapshot:  marshaled,
+		}
+
+		if err := c.Proposals.Upsert(&p); err != nil {
+			return proposalIds, err
+		}
+		proposalIds = append(proposalIds, proposal.ID)
+	}
+
+	return proposalIds, nil
+}
+
+func (c *FetchForSpace) fetchProposalVotes(ctx context.Context, id string) error {
+
+	opts := []snapshot.ListVotesOption{
+		snapshot.ListVotesWithOrderBy("created", client.OrderDirectionAsc),
+		snapshot.ListVotesWithProposalIDsFilter(id),
+	}
+
+	offset := 0
+	for {
+		votes, err := c.fetchVotes(ctx, offset, opts)
+		if errors.Is(err, snapshot.ErrTooManyRequests) {
+			log.Warn().Err(err).Msg("snapshot api limits are reached")
+			<-time.After(5 * time.Second)
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		if err := c.processVotes(votes); err != nil {
+			return err
+		}
+
+		if len(votes) < votesPerRequest {
+			err := c.Proposals.MarkVotesProcessed(id)
```

Comment:
Do not mark active proposal to avoid missing votes in the future

Reply:
corrected.